backend/dal/Services/Admin/Concrete/ClaimService.cs

```diff
@@ -0,0 +1,134 @@
+        /// <param name="id"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public Claim Get(Guid id)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.AdminRoles);
+
+            return this.Context.Claims.AsNoTracking().FirstOrDefault(u => u.Id == id) ?? throw new KeyNotFoundException();
+        }
+
+        /// <summary>
+        /// Get the claim with the specified name.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public Claim GetByName(string name)
+        {
+            return this.Context.Claims.AsNoTracking().FirstOrDefault(r => r.Name == name) ?? throw new KeyNotFoundException();
+        }
+
+        /// <summary>
+        /// Updates the specified claim in the datasource.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        /// <returns></returns>
+        public override Claim Update(Claim entity)
+        {
+            entity.ThrowIfNull(nameof(entity));
+            this.User.ThrowIfNotAuthorized(Permissions.AdminRoles);
+
+            var claim = this.Context.Claims.Find(entity.Id) ?? throw new KeyNotFoundException();
+
+            this.Context.Entry(claim).CurrentValues.SetValues(entity);
+            base.Update(claim);
+            this.Context.Detach(claim);
+            return claim;
+        }
+
+        /// <summary>
+        /// Remove the specified claim from the datasource.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <exception type="KeyNotFoundException">Entity does not exist in the datasource.</exception>
+        public override void Remove(Claim entity)
+        {
+            entity.ThrowIfNull(nameof(entity));
+            this.User.ThrowIfNotAuthorized(Permissions.AdminRoles);
+
+            var claim = this.Context.Claims.Find(entity.Id) ?? throw new KeyNotFoundException();
+
+            this.Context.Entry(claim).CurrentValues.SetValues(entity);
+            base.Remove(claim);
+        }
+
+        /// <summary>
+        /// Remove the claims from the datasource, excluding those listed.
+        /// </summary>
+        /// <param name="exclude"></param>
+        /// <returns></returns>
+        public int RemoveAll(Guid[] exclude)
+        {
+            this.User.ThrowIfNotAuthorized(Permissions.AdminRoles);
+            var claims = this.Context.Claims
+                .Include(r => r.Roles)
+                .Where(r => !exclude.Contains(r.Id));
+            claims.ForEach(r =>
+            {
+                r.Roles.Clear();
+            });
+
+            this.Context.Claims.RemoveRange(claims);
+            var result = this.Context.CommitTransaction();
```

Comment:
I would appreciate this being typed to an int for clarity instead of a var.

Reply:
Not sure what you mean here.  `var` is just a syntax feature that allows us to just inherit the type of the right.  Which is an int.

Reply:
yeah, this is more of a nitpick, it isn't immediately obvious to me what the return type of CommitTransaction is in this case, I'd need to read the doc of the method.

For me that is usually when I decide to use the actual type instead of var.

Reply:
Agree with Devin